src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -232,17 +240,49 @@ public PackageInfo BuildModel(string id, XElement element)
             });
         }
 
+        private void EnsureValidFindPackagesResponse(XDocument doc)
+        {
+            if (!string.Equals(doc.Root.GetDefaultNamespace().ToString(), _defaultNamespace,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                throw new FormatException("The XML is not a valid NuGet v2 service response to FindPackagesById() request");
```

Comment:
As per MSDN "The exception that is thrown when the format of an argument is invalid, or when a composite format string is not well formed." The format here is Xml and because the file could be loaded into XDocument the it is a valid Xml document so the format is correct. If we had an XSD (hopefully we don't) for this Xml file we would have thrown an XmlValidationException because the document is not valid according to the schema but this is not format exception. 
`InvalidOperationException` is probably general enough that you can throw it almost everywhere where you can't find a better exception and you don't want to create a specialized exception.